app/main.py

```diff
@@ -1,13 +1,44 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
     pass
 
-
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {"errors":
+        [{
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"} for error in errors],
+    "path": file_path,
+    "status": "passed"
+    if errors == []
+    else "failed"
+    }
```

Comment:
It seems like you're duplicating the code from `format_linter_error` function inside `format_single_linter_file` function. Instead, you should use `format_linter_error` function to format each error in the list. This would make your code more DRY and easier to maintain.